source/PlayerInfo.cpp

```diff
@@ -841,15 +841,32 @@ void PlayerInfo::AddShip(const shared_ptr<Ship> &ship)
 
 
 
-// Buy a ship of the given model, and give it the given name.
-void PlayerInfo::BuyShip(const Ship *model, const string &name)
+// Charge the player the cost of a ship and give it the specified name:
+void PlayerInfo::BuyShip(const Ship *model, const std::string &name)
+{
+	ReceiveShip(model, name, true, false);
+}
+
+
+
+// Receive a ship as a gift
+void PlayerInfo::GiftShip(const Ship *model, const std::string &name)
+{
+	return ReceiveShip(model, name, 0, true);
+}
+
+
+
+// Purchase or receive a gift of a ship of the given model and name
+void PlayerInfo::ReceiveShip(const Ship *model, const string &name, int64_t cost, bool gifting)
```

Comment:
Someone still didn't double check what I posted.
```suggestion
void PlayerInfo::BuyShip(const Ship *model, const std::string &name)
{
	ReceiveShip(model, name, false);
}



// Receive a ship as a gift
void PlayerInfo::GiftShip(const Ship *model, const std::string &name)
{
	ReceiveShip(model, name, true);
}



// Purchase or receive a gift of a ship of the given model and name
void PlayerInfo::ReceiveShip(const Ship *model, const string &name, bool gifting)
```